backend/src/main/java/apsi/team3/backend/controller/EventController.java

```diff
@@ -76,14 +85,51 @@ public ResponseEntity<EventDTO> createEvent(@RequestParam(name = "image", requir
         }
     }
 
-    @PutMapping("/{id}")
-    public ResponseEntity<EventDTO> replaceEvent(@PathVariable("id") Long id, @RequestBody EventDTO eventDTO) throws ApsiValidationException {
-        validateSameId(id, eventDTO);
-        if (eventService.notExists(id)) {
-            return ResponseEntity.notFound().build();
+    @PutMapping(value="/{id}", consumes = { MediaType.MULTIPART_FORM_DATA_VALUE })
+    public ResponseEntity<EventDTO> replaceEvent(
+            @PathVariable("id") Long id,
+            @RequestPart("event") String event,
+            @RequestPart(name = "image", required = false) MultipartFile image
+    ) throws ApsiValidationException {
+        if (image != null && image.getSize() > 500_000)
+            throw new ApsiValidationException("Zbyt duży obraz. Maksymalna wielkość to 500 KB", "image");
+
+        try {
+            var mapper = new ObjectMapper();
+            mapper.registerModule(new JavaTimeModule());
+            var eventDTO = mapper.readValue(event, EventDTO.class);
+
+            validateSameId(id, eventDTO);
+            var oldEvent = eventService.getEventById(id);
+            if (oldEvent.isEmpty()) {
+                return ResponseEntity.notFound().build();
+            }
+
+            var resp = eventService.replace(eventDTO, image);
+
+            boolean timeChanged = Objects.equals(eventDTO.getStartTime(), oldEvent.get().getStartTime())
```

Comment:
żeby ta zmienna przechowywała to co sugeruje nazwa, to to wyrażenie powinno być zanegowane. Tylko, że wtedy nie będzie działać (teraz działa), bo eventDTO.getStartTime() oraz oldEvent.get().getStartTime() oba z jakiegoś powodu zwracają null (sprawdzałem).